aptos-move/aptos-vm-types/src/change_set.rs

```diff
@@ -921,11 +774,127 @@ impl VMChangeSet {
     }
 }
 
+/// Builds a new change set from the storage representation.
+///
+/// **WARNING**: this creates a write set that assumes dynamic change set optimizations to be disabled.
+/// this needs to be applied directly to storage, you cannot get appropriate reads from this in a
+/// dynamic change set optimization enabled context.
+/// We have two dynamic change set optimizations, both there to reduce conflicts between transactions:
+///  - exchanging delayed fields and leaving their materialization to happen at the end
+///  - unpacking resource groups and treating each resource inside it separately
+///
+/// **WARNING**: Has complexity O(#write_ops) because we need to iterate
+/// over blobs and split them into resources or modules. Only used to
+/// support transactions with write-set payload.
+///
+/// Note: does not separate out individual resource group updates.
+pub fn create_vm_change_set_with_module_write_set_when_delayed_field_optimization_disabled(
+    change_set: StorageChangeSet,
```

Comment:
Was wondering what are the mock you are trying to do with this trait? I would avoid introducing a new trait just for one single instantiation.

Reply:
Do you mean change set interface? We need to run the checks for both change set without modules, and new `userSessionChangeSet` with modules. Conceptually, we can also do it for storage change set, so it's more about the functionality: we need to query write op info + events

Reply:
Got it. Sounds good to me then.